pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
+		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
```

Comment:
@mfojtik - You have a quota in this project as I can see, but did you have a LimitRange?  

Setting requests alone will not cause the `ClusterResourceOverride` admission handler to scale the container resource limits.  `ClusterResourceOverride` works in the opposite direction of scaling requests based on user supplied limits.

Your quota in this case is saying a limit must be specified, so you either need a LimitRange in your project that defines a default limit.